### PR TITLE
Redesign "Getting Started" page to be less text heavy

### LINKED ISSUE #18: "Getting Started" page should be less text heavy
The "Getting Started" page should be prettier and more user-friendly than just a giant wall of text.

### PR DESCRIPTION
**Pull request by allejo**
The previous "Getting Started" page was very text heavy and was easy to get lost in. This PR takes the same content (like ~98% of it) and restructures it to make it easier to follow; additionally, some new things have been added in.

The only left to add in is a jumbotron image but everything else is ready to be reviewed. If anyone can or wants to provide feedback on this page, I'd really appreciate it!

Closes #18

## Tasks

- [x] Port "Registration" section
- [x] Port "Teams" section
- [x] Port "Game Modes" section
- [x] Port "Flags" section
- [x] Move CSS keyboard into a reusable include or macro
- [x] Set up photoshoot for "Super Flags" section
  - [x] Super flags
  - ~~Team flags~~
- [x] Add jumbotron image
- [x] Port "Controls" section
  - [x] Add arrow keys for movement
  - [x] Add mouse controls + mouse icon
- [x] Port "User Interface" section
- [x] Rewrite the "heading group" (i.e. the `<p> + <h3>` combo) into a component instead of reusing the same utility classes
- [x] Rewrite page splits as a component (i.e. the `<h2>` with y-axis border)
- [x] Fix standalone special keys not sizing correctly
- [x] Split up sections into partial files because this page is getting too hard to manage
- [x] Add table of contents at the top of the page
- [x] Fix spacing and alignment of things
  - [x] Fix placement of controls
  - [x] Make everything mobile responsive
- [x] Add link to Navigation bar
- [x] Remove markdown version of "Getting Started"

## Page Preview

For anyone who can't build the site locally, here's a preview of what the new Getting Started page looks like.

getting-started-new

### CODE REVIEW

File: _pages-partials/getting-started/05-controls.html.twig [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+                                </kbd>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+                <div class="col-sm-6 col-lg-4 mb-4">
+                    <div class="kb-control-group">
+                        <p class="title">
+                            Identify / Lock On
+                            <abbr title="with Guided Missile flag">*</abbr>
```
Comment (allejo) [inline _pages-partials/getting-started/05-controls.html.twig]: I'm abusing `<abbr>` here, let's make this a `<span>` with a `title` and a class to style it like an `<abbr>`.

File: _pages-partials/getting-started/05-controls.html.twig [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+                            </div>
+                        </div>
+                    </div>
+                </div>
+                <div class="col-sm-6 col-lg-4">
+                    <div class="kb-control-group">
+                        <p class="title">Drop Flag</p>
+                        <div class="controls">
+                            <div class="control">
+                                <kbd class="c-keyboard__key space" data-align="center">space</kbd>
```
Comment (allejo) [inline _pages-partials/getting-started/05-controls.html.twig]: Can this become a Twig macro to generate KB keys instead? It'd be nice to generate keys consistently (especially if they have "shift values").

**Comment by tainn1** [comment]
This looks quite amazing, @allejo. If anything, for now, I would just like to share some grammatical preferences, though I could be wrong here, so please, correct me if so.

New additions are in **bold**, so be wary of cases where text is added, not replaced, and the cases of bold commas (**,**).

Register a Callsign
> Some servers provide additional benefits for registered users, such as being ~~allows~~ **allowed** to vote in polls.

Capture the Flag (CTF)
> In CTF, each of the four main teams can have one or more team ~~flag~~ **flags** and one or more team ~~base~~ **bases**, both colored to match their team.

Super Flags
> View the flags documentation for a list of all official flags and their ~~description~~ **descriptions**.

Scoreboard
> The scoreboard consists of three main ~~section~~ **sections**:

Team Scores
> CTF: The amount of captures a team has made vs the amount of times the team ~~was captured~~ **had their own flag captured** *# potentially unnecessary, but a team being captured sounds off*

Radar
> One of the game's most powerful features is an all-seeing radar**,** allowing you **to** see every tank, shot, and object on the playing field.

Control Panel
> This component displays ~~every thing~~ **everything** that is happening on the server**,** including:

File: _sass/abstracts/_variables.scss [hunk 3]
```diff
@@ -50,12 +50,13 @@ $spacing_directions: (
...
-    h4: 22px,
-    h5: 18px,
-    h6: 16px
+    h1: em(34px),
+    h2: em(30px),
+    h3: em(26px),
+    h4: em(22px),
+    h5: em(18px),
+    h6: em(16px),
+    small: 0.85em,
```
[the remark this answers is not in the document]
Comment (allejo) [reply _sass/abstracts/_variables.scss]: So I've defined an `em()` function in Sass that will convert `px` into `em` based values. For me, it's just easier to estimate font sizes based on pixel values but the compiled CSS will have `em` values.

File: _pages-partials/getting-started/05-controls.html.twig [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+                        </div>
+                    </div>
+                </div>
+                <div class="col-sm-6 col-lg-4 mb-4">
+                    <div class="kb-control-group">
+                        <p class="title">
+                            Identify / Lock On
+                            <span class="tooltip" title="with Guided Missile flag">*</span>
+                            player
+                            <span class="sr-only">with Guided Missile flag</span>
```
[the remark this answers is not in the document]
Comment (allejo) [reply _pages-partials/getting-started/05-controls.html.twig]: I'd have to check how screen readers will read the asterisk and its title (probably should do an `aria-hidden` on the asterisk then). But the `sr-only` span is text that'll only be read by screen readers. Sighted users will see the asterisk and be able to hover over it to see that message.

File: _pages-partials/getting-started/01-register-callsign.html.twig [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<section class="row">
+    <div class="col-lg-7">
+        <header class="c-hgroup">
+            <p class="c-hgroup__subtitle">What's in a name?</p>
+            <h2 id="register-a-callsign" class="c-hgroup__title">Register a Callsign</h2>
+        </header>
+
+        <p class="mb-4">
+            BZFlag allows players to register their name, also known as their callsign. Some servers provide additional benefits for registered users, such as being allowed to vote in polls. Another benefit is that players can not use another player's registered callsign. To register a callsign, visit our forums and register an account. The username and password on the forums will be used as callsign and password in the game.
```
[the remark this answers is not in the document]
Comment (allejo) [reply _pages-partials/getting-started/01-register-callsign.html.twig]: I don't *think* we've covered the difference between global and local auth anywhere but I'd favor talking about that somewhere in the documentation pages. This page is intended for new players and they shouldn't need to care about global vs local auth.

File: _pages-partials/getting-started/05-controls.html.twig [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+                        <div class="col-sm-6 text-center">
+                            <div class="kb-control-group">
+                                <p class="title">
+                                    Send Chat to <span class="tooltip" title="last person that was messaged by or messaged you">Recipient</span>
+                                </p>
+                                <div class="controls">
+                                    <div class="control">
+                                        {{ keyboard_key('.') }}
+                                    </div>
+                                </div>
```
[the remark this answers is not in the document]
Comment (allejo) [reply _pages-partials/getting-started/05-controls.html.twig]: I like adding `W`.

Switching the tabs is mentioned at the bottom of the page in the "User Interface" section under "Control Panel". Do you think that should belong in this section instead?

[the remark this answers is not in the document]
Comment (jwmelto) [reply _pages-partials/getting-started/05-controls.html.twig]: I think it was one of the attempted re-designs of the HUD where I realized that there were actually different tabs, but I'd been using the function keys to toggle what messages were shown for years. It's all a matter of style; if you feel it's covered adequately elsewhere, that's fine.

File: _pages-partials/getting-started/03-game-modes.html.twig [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            <article>
+                <header>
+                    <img src="{{ url('/assets/images/getting-started/rabbit-hunters.png') }}"
+                         alt="Two hunter tanks pursuing a rabbit tank"
+                         class="img-fluid mb-3"
+                    >
+
+                    <div class="c-hgroup">
+                        <div class="c-hgroup__subtitle">It's always rabbit season</div>
+                        <h3 id="rabbit-chase" class="c-hgroup__title h2">Rabbit Chase</h3>
```
[the remark this answers is not in the document]
Comment (allejo) [reply _pages-partials/getting-started/03-game-modes.html.twig]: Man pages call it "rabbit hunt" so "rabbit hunt" it is!

File: _pages-partials/getting-started/05-controls.html.twig (hunk 4 above)
[the remark this answers is not in the document]
Comment (allejo) [reply _pages-partials/getting-started/05-controls.html.twig]: VoiceOver reads it fine, I'll hope JAWS does the same.

**Comment by allejo** [comment]
Solely because I'm impatient, I'm merging this branch. Any further feedback should be created as a separate issue.

**Pull request merged**